components/public-api/typescript/src/metrics.ts

```diff
@@ -283,6 +293,12 @@ export class MetricsReporter {
         if (!enabled) {
             return;
         }
+        while (this.pendingRequests.length) {
+            const request = this.pendingRequests.shift();
+            if (request) {
+                this.doPost(request);
```

Comment:
WDYT about putting an `await` here, as a measure to not overload the backend with too many errors at once?

Reply:
in theory we should also aggregate counters and histograms instead of pushing so many requests